TRACING.md

```diff
@@ -0,0 +1,312 @@
+
+     - The [Tracing options](#tracing-options) section.
+
+     - If you also want full debug and guest journal redirection, see the
+       [Enable full agent debug](#enable-full-agent-debug) section.
+
+1. Install and run Jaeger on the host.
+
+   The simplest method to start using Jaeger is to follow the instructions to run the
+   ["all-in-one" Docker\* image](https://www.jaegertracing.io/docs/1.8/getting-started/#all-in-one).
+
+1. Start the VSOCK Jaeger redirector script on the host:
+
+   ```
+   $ sudo scripts/vsock-to-udp-server.sh
+   ```
+
+1. Start the journal redirector script on the host (OPTIONAL):
+
+   ```
+   $ sudo scripts/vsock-to-text-server.sh
+   ```
+
+1. Create a Kata Container.
+
+1. Stop the Kata Container.
+
+1. View the [Jaeger UI](https://www.jaegertracing.io/docs/getting-started).
+
+# Requirements
+
+## Host environment
+
+- Host kernel must support the VSOCK socket type:
+
+  kernel config option: `CONFIG_VHOST_VSOCK`.
+
+- VSOCK kernel module must be loaded:
+
+   ```
+   $ sudo modprobe vhost_vsock
+   ```
+
+- VSOCK support must be enabled in the `kata-runtime`'s `configuration.toml`
+  config file:
+
+  ```
+  $ sudo sed 's/#use_vsock =.*/use_vsock = true/g' /usr/share/defaults/kata-containers/configuration.toml
+  ```
+
+## Guest environment
+
+- Guest kernel must support the VSOCK socket type:
+
+  kernel config option: `CONFIG_VIRTIO_VSOCKETS`.
+
+# Limitations
+
+- The image configured in the `kata-runtime` `configuration.toml` file must
+  be specified using `image=` (`initrd=` is not supported).
+
+  This is due to having to run additional systemd services inside the VM
+  currently, to proxy the Jaeger trace flows out to the Jaeger collector running
+  on the host.
+
+  See the
+  [Guest to Host communication using VSOCK](#guest-to-host-communication-using-vsock)
+  section for further details.
+
+- Tracing is only finalised when the workload has exited
+
+  Although trace *can* be inspected before the workload exits, it will be
+  incomplete. This is shown as `<trace-without-root-span>` in the Jaeger UI.
+
+  If the workload is still running, the trace transaction, which spans the entire
+  runtime of the `kata-agent` will not have been completed. To view the complete
+  trace details, first stop the Kata Container.
+
+  See the [VM shutdown](#vm-shutdown) section for further details.
+
+# Appendix: Setting up a tracing development environment
+
+If you wish to debug, further develop or test tracing, follow the steps below
+to create a more useful environment:
+
+## Set up a debug console
+
+This allows direct shell access to the container.
+
+See
+[the developer guide](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#set-up-a-debug-console)
+for further details.
+
+## Guest journal redirector
+
+Since agent tracing requires VSOCK, and enabling VSOCK in `configuration.toml`
+disables the `kata-proxy` (since it is not required), the Kata Container has
```

Comment:
Lines 291-294 suggested rewrite: disables the 'kata-proxy' (not required), the Kata Container has no way to display the guest OS journal messages, which are normally redirected through the proxy. This makes debugging difficult. To overcome this issue, create another pair of ...